### PR TITLE
Dependency Updates

### LINKED ISSUE #48: Update log4j to fix CVE-2021-44228
This project uses log4j versions potentially affected by [CVE-2021-44228](https://nvd.nist.gov/vuln/detail/CVE-2021-44228):

```
[INFO]    org.apache.logging.log4j:log4j-api:jar:2.11.1:test
```
Dependency tree:
```
[INFO] +- org.elasticsearch.client:elasticsearch-rest-high-level-client:jar:7.15.0:test
[INFO] |  +- org.elasticsearch:elasticsearch:jar:7.15.0:test
[INFO] |  |  +- org.elasticsearch:elasticsearch-core:jar:7.15.0:test
[INFO] |  |  +- org.elasticsearch:elasticsearch-secure-sm:jar:7.15.0:test
[INFO] |  |  +- org.elasticsearch:elasticsearch-x-content:jar:7.15.0:test
[INFO] |  |  |  +- org.yaml:snakeyaml:jar:1.26:test
[INFO] |  |  |  +- com.fasterxml.jackson.core:jackson-core:jar:2.10.4:test
[INFO] |  |  |  +- com.fasterxml.jackson.dataformat:jackson-dataformat-smile:jar:2.10.4:test
[INFO] |  |  |  +- com.fasterxml.jackson.dataformat:jackson-dataformat-yaml:jar:2.10.4:test
[INFO] |  |  |  \- com.fasterxml.jackson.dataformat:jackson-dataformat-cbor:jar:2.10.4:test
[INFO] |  |  +- org.elasticsearch:elasticsearch-geo:jar:7.15.0:test
[INFO] |  |  +- org.apache.lucene:lucene-core:jar:8.9.0:test
[INFO] |  |  +- org.apache.lucene:lucene-analyzers-common:jar:8.9.0:test
[INFO] |  |  +- org.apache.lucene:lucene-backward-codecs:jar:8.9.0:test
[INFO] |  |  +- org.apache.lucene:lucene-grouping:jar:8.9.0:test
[INFO] |  |  +- org.apache.lucene:lucene-highlighter:jar:8.9.0:test
[INFO] |  |  +- org.apache.lucene:lucene-join:jar:8.9.0:test
[INFO] |  |  +- org.apache.lucene:lucene-memory:jar:8.9.0:test
[INFO] |  |  +- org.apache.lucene:lucene-misc:jar:8.9.0:test
[INFO] |  |  +- org.apache.lucene:lucene-queries:jar:8.9.0:test
[INFO] |  |  +- org.apache.lucene:lucene-queryparser:jar:8.9.0:test
[INFO] |  |  +- org.apache.lucene:lucene-sandbox:jar:8.9.0:test
[INFO] |  |  +- org.apache.lucene:lucene-spatial-extras:jar:8.9.0:test
[INFO] |  |  +- org.apache.lucene:lucene-spatial3d:jar:8.9.0:test
[INFO] |  |  +- org.apache.lucene:lucene-suggest:jar:8.9.0:test
[INFO] |  |  +- org.elasticsearch:elasticsearch-cli:jar:7.15.0:test
[INFO] |  |  |  \- net.sf.jopt-simple:jopt-simple:jar:5.0.2:test
[INFO] |  |  +- com.carrotsearch:hppc:jar:0.8.1:test
[INFO] |  |  +- org.lz4:lz4-java:jar:1.8.0:test
[INFO] |  |  +- joda-time:joda-time:jar:2.10.10:test
[INFO] |  |  +- com.tdunning:t-digest:jar:3.2:test
[INFO] |  |  +- org.hdrhistogram:HdrHistogram:jar:2.1.9:test
[INFO] |  |  +- org.apache.logging.log4j:log4j-api:jar:2.11.1:test    <=====
[INFO] |  |  +- org.elasticsearch:jna:jar:5.7.0-1:test
[INFO] |  |  \- org.elasticsearch:elasticsearch-plugin-classloader:jar:7.15.0:test
```

### PR DESCRIPTION
**Pull request by jakobbraun**
Closes #48

### CODE REVIEW

File: doc/changes/changes_2.0.3.md [hunk 1]
```diff
@@ -0,0 +1,31 @@
+# Virtual Schema for ElasticSearch 2.0.3, released 2021-12-13
```
Comment (kaklakariada) [inline doc/changes/changes_2.0.3.md:1]:
```suggestion
# Virtual Schema for ElasticSearch 2.0.3, released 2021-12-14
```

File: doc/changes/changes_2.0.3.md [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+
+Code name: Dependency Updates
+
+## Summary
+
+In this release we updated the project's dependencies. By that we fixed the transitive dependency CVE-2021-44228 in log4j. The vulnerability only affected test code.
```
Comment (kaklakariada) [inline doc/changes/changes_2.0.3.md:7]:
```suggestion
In this release we updated the project's dependencies. By that we fixed the transitive dependency [CVE-2021-44228](https://nvd.nist.gov/vuln/detail/CVE-2021-44228) in log4j. The vulnerability only affected test code.
```

File: pom.xml [hunk 3]
```diff
@@ -113,25 +113,25 @@
...
         </dependency>
         <!-- Overrides Junit 4.12 which is used in test containers but contains
-                a security issue: https://ossindex.sonatype.org/vuln/7ea56ad4-8a8b-4e51-8ed9-5aad83d8efb1 -->
+                    a security issue: https://ossindex.sonatype.org/vuln/7ea56ad4-8a8b-4e51-8ed9-5aad83d8efb1 -->
         <dependency>
             <groupId>junit</groupId>
             <artifactId>junit</artifactId>
             <version>4.13.2</version>
             <scope>test</scope>
         </dependency>
```
Comment (kaklakariada) [inline pom.xml:132]: This should not be necessary any more, testcontainers has updated to the current junit version.

**Review by kaklakariada — CHANGES_REQUESTED** [review]

File: pom.xml [hunk 4]
```diff
@@ -138,6 +131,14 @@
...
+
+        <!-- Overrides log4j-api 2.11.1 which is used in elasticsearch-rest-high-level-client but contains a security issue: https://nvd.nist.gov/vuln/detail/CVE-2021-44228 -->
+        <dependency>
+            <groupId>org.apache.logging.log4j</groupId>
+            <artifactId>log4j-api</artifactId>
+            <version>2.16.0</version>
+            <scope>test</scope>
+        </dependency>
```
Comment (jakobbraun) [inline pom.xml:141]: Strange. I checked with the ossindex plugin and it did not find a vulnerability.
On main, it does find the CVE.

**Review by kaklakariada — APPROVED** [review]

**Pull request merged**